packages/go_router/lib/src/information_provider.dart

```diff
@@ -109,7 +105,7 @@ class GoRouteInformationProvider extends RouteInformationProvider
     final bool replace;
     switch (type) {
       case RouteInformationReportingType.none:
-        if (_valueInEngine.location == routeInformation.location &&
+        if (_valueInEngine.uri.path == routeInformation.uri.path &&
```

Comment:
This would also need to check the query parameter and hash fragment

Reply:
So something like
```dart
if (_valueInEngine.uri.path == routeInformation.uri.path &&
            _valueInEngine.uri.queryParameters ==
                routeInformation.uri.queryParameters &&
            _valueInEngine.uri.fragment == routeInformation.uri.fragment &&
            const DeepCollectionEquality()
                .equals(_valueInEngine.state, routeInformation.state))
```
?

Reply:
you probably need to do collection equality for query parameter

Reply:
I believe I have done these correctly, I went ahead and added DeepCollectionEquality checks to the `queryParameters` and `fragment`s, as well as the `path`s and `state`s.